src/XMakeBuildEngine/Evaluation/Evaluator.cs

```diff
@@ -816,11 +816,42 @@ private static ProjectTargetInstance ReadNewTargetElement(ProjectTargetElement t
             string endPass2 = String.Format(CultureInfo.CurrentCulture, "Evaluate Project {0} - End Pass 2 (Item Definitions)", projectFile);
             DataCollection.CommentMarkProfile(8818, endPass2);
 #endif
+            LazyItemEvaluator<P, I, M, D> lazyEvaluator = null;
+            lazyEvaluator = new LazyItemEvaluator<P, I, M, D>(_data, _itemFactory, _buildEventContext, _loggingService);
+
             // Pass3: evaluate project items
             foreach (ProjectItemGroupElement itemGroupElement in _itemGroupElements)
             {
-                EvaluateItemGroupElement(itemGroupElement);
+                EvaluateItemGroupElement(itemGroupElement, lazyEvaluator);
+            }
+
+            if (lazyEvaluator != null)
```

Comment:
It's a poor man's feature flag.  Comment out the code that creates the LazyEvaluator, and it will fall back to the old logic.